tests/unit/jobserver/views/test_workspaces.py

```diff
@@ -488,6 +484,37 @@ def get_repo_is_private(self, owner, name):
     assert response.context_data["show_publish_repo_warning"]
 
 
+def test_workspacedetail_authorized_private_repo_show_workspace_admin_panel(
+    rf, project_membership, role_factory
+):
+    project = ProjectFactory()
+
+    # a workspace with a "private" repo
+    workspace = WorkspaceFactory(
+        project=project, repo=RepoFactory(url="http://example.com/repo/private1")
+    )
```

Comment:
thought: Without actually trying anything out, I *think* that regardless of the repository being private or public, it shouldn't make any difference to whether `show_admin` is enabled — which is what we're testing. That is, the outcome of the test is now primarily based on the user's permissions.

Where we *would* see a difference between a private and public repository is in the rendered template.

(I'm not too sure how much we favour testing content of rendered templates. Though, we do test a rendered template elsewhere in these tests though.)